packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -1477,16 +1480,16 @@ describe('Select', () => {
             });
 
             it('emits filter-input event with empty filterText when dropdown is closed', async () => {
-                const filterInputEvent = jasmine.createSpy();
-                element.addEventListener('filter-input', filterInputEvent);
+                const filterInputEvent = jasmine.createSpy<SelectFilterInputEventHandler>();
+                element.addEventListener(
+                    'filter-input',
+                    filterInputEvent as unknown as EventListener
+                );
                 await pageObject.openAndSetFilterText('o');
                 await pageObject.clickAway();
                 expect(filterInputEvent).toHaveBeenCalledTimes(2);
                 expect(
-                    (
-                        (filterInputEvent.calls.argsFor(1)[0] as CustomEvent)
-                            .detail as SelectFilterInputEventDetail
-                    ).filterText
+                    filterInputEvent.calls.argsFor(1)[0].detail.filterText
```

Comment:
This is much better. We lie to addEventListener about the handler to tradeoff for the spy getting richer type info. We may be able to create a feature request in jasmine to make createSpy easier to use with addEventListener but not going down that path for now.